wafer/schedule/admin.py

```diff
@@ -151,41 +143,76 @@ def prefetch_schedule_items():
                 .all())
 
 
+def prefetch_slots():
+    return list(Slot.objects.all())
+
+
+# Validators are listed as (function, error type, error message) tuples
+SLOT_VALIDATORS = []
+SCHEDULE_ITEM_VALIDATORS = []
+
+
+# Helpers for people extending the tests
+def register_slot_validator(function, err_type, msg):
+    global SLOT_VALIDATORS
```

Comment:
Given that we're calling `.append(...)` we don't actually need the `global`s here but I guess it's not a bad idea to make it clear that these are globals. Happy either way, just checking intent.